mergin/client.py

```diff
@@ -365,25 +365,31 @@ def apply_pull_changes(self, changes, temp_dir):
                 # special care is needed for geodiff files
                 if self.is_versioned_file(path) and k == 'updated':
                     if path in modified:
+                        f_server_backup = self.fpath(f'{path}-server_backup', temp_dir)
                         server_diff = self.fpath(f'{path}-server_diff', temp_dir)  # single origin diff from 'diffs' for use in rebase
                         rebased_diff = self.fpath(f'{path}-rebased', temp_dir)
-                        patchedfile = self.fpath(f'{path}-patched', temp_dir)  # patched server version with local changes
-                        changeset = self.fpath(f'{path}-local_diff', temp_dir)  # final changeset to be potentially committed
+                        shutil.copy(src, f_server_backup)  # temporary backup of file pulled from server for test and recovery
```

Comment:
copying may be dangerous here... if the 'src' is open (and there are -wal files around), we may loose some data which are not committed to the .gpkg yet. Ideally all operations with .gpkg should be going through geodiff and if some manual copying is impossible to avoid, maybe at least we should have a geodiff function to flush and possible pending changes from WAL to gpkg.